### DISCUSSION TITLE
Choosing the correct service for a door buzzer

### DISCUSSION THREAD

**Discussion by etkaar** [HomeSpan/HomeSpan#807] [General]
What would be the best fitting [service](https://github.com/HomeSpan/HomeSpan/blob/master/docs/ServiceList.md) for a door buzzer? ☺️ The buzzer must be held down as long someone wants to open a door, so it is not a click buzzer. I want it to be automatically held down for about 2 seconds.

**Comment by HomeSpan** [comment]
It depends what you want HomeKit to do with the buzzer.   Most people use the Doorbell Service in conjunction with the LockMechanism Service.  The Doorbell Service allows a ring on your doorbell to be sent to HomeKit which trigger a notification on your iPhone and can also play a chime tone on any HomePods.  The LockMechanism then allows you to open a lock when someone rings the door bell.

Since HomeKit can't automatically unlock anything, I'm not sure the 2-seconds really comes into play.

**Reply by etkaar** [reply]
I used the wrong term, sorry. What I meant is the **electric door opener**. It makes a loud noise and during that the door can be opened; after that, the door can't be opened. So it is not a clicky one.

Which service do you think would be the correct one for that?

**Comment by HomeSpan** [comment]
Did you see my response under https://github.com/HomeSpan/HomeSpan/discussions/809#discussioncomment-8838331 (which seems to be the same topic, correct)?   You want to use a pushbutton construct as fully detailed in the EmulatedPushButton tutorial to accomplish what I think you are trying to do.
